src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -194,19 +206,33 @@ public Integer invoke(File f, VirtualChannel channel) throws IOException, Interr
         static final StatusCheck STATUS_CHECK_INSTANCE = new StatusCheck();
 
         @Override public Integer exitStatus(FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
+            return exitStatus(workspace, listener);
+        }
+
+        /**
+         * Like {@link #exitStatus(FilePath, Launcher, TaskListener)} but not requesting a {@link Launcher}, which would not be available in {@link #watch} mode anyway.
+         */
+        protected Integer exitStatus(FilePath workspace, TaskListener listener) throws IOException, InterruptedException {
```

Comment:
Add CheckForNull so nobody unboxes it to `int` with FindBugs being silent?

Reply:
Could do that, but actually I think I want to mark this

```java
@Restricted(ProtectedExternally.class)
```

as a reminder that only this class is expected to be calling it.

Reply:
Never mind—plain `protected` already accomplishes that. Can annotate nullness.